File: Documentation/5.4/Samples/python/ClientApi/Operations/Maintenance/Indexes/DeleteIndex.py
from ravendb import DeleteIndexOperation
from ravendb.documents.operations.definitions import VoidMaintenanceOperation

from examples_base import ExampleBase


class DeleteIndex(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_delete_index(self):
        with self.embedded_server.get_document_store("DeleteIndex") as store:
            self.add_index_orders_totals(store)
            # region delete_index
            # Define the delete index operation, specify the index name
            delete_index_op = DeleteIndexOperation("Orders/Totals")

            # Execute the operation by passing it to maintenance.send
            store.maintenance.send(delete_index_op)
            # endregion


class Foo:
    # region syntax
    class DeleteIndexOperation(VoidMaintenanceOperation):
        def __init__(self, index_name: str): ...

    # endregion

File: Documentation/5.4/Samples/php/ClientApi/Operations/Maintenance/Indexes/DeleteIndex.php
<?php

namespace RavenDB\Samples\ClientApi\Operations\Maintenance\Indexes;

use RavenDB\Documents\DocumentStore;
use RavenDB\Documents\Operations\Indexes\DeleteIndexOperation;

class DeleteIndex
{
    public function samples(): void
    {
        $store = new DocumentStore();
        try {
            # region delete_index
            // Define the delete index operation, specify the index name
            $deleteIndexOp = new DeleteIndexOperation("Orders/Totals");

            // Execute the operation by passing it to Maintenance.Send
            $store->maintenance()->send($deleteIndexOp);
            # endregion
        } finally {
            $store->close();
        }
    }
}

/*
interface IFoo
{
    # region syntax
    DeleteIndexOperation(?string $indexName)
    # endregion
}
*/
